third_party/purge/ext.purge.php

```diff
@@ -100,7 +103,19 @@ public function send_purge_request()
 		
 		foreach ($urls as $url)
 		{
-			send_purge_request($url, $this->port);
+			//now loop through urls for this channel
+			$this->EE->db->select('*');
+			$this->EE->db->where('channel_id',(int) $meta['channel_id']);
+			$channelPatterns = $this->EE->db->get_where('purge_rules')->result_array();
+			
+			foreach($channelPatterns as $pattern)
```

Comment:
I'm not really a fan of changing the behavior of the hook on update to no longer purge the whole site. They'll upgrade the add-on then it will appear not to be working. Can you revert to the old behavior if there are no channel patterns saved?